src/components/Alert.tsx

```diff
@@ -0,0 +1,130 @@
+};
+
+const getColorByType = (type: string, isDarkTheme?: boolean) => {
+  if (isDarkTheme) {
+    switch (type) {
+      case 'success': return '#B5FFD9';
+      case 'info': return '#90E0EF';
+      case 'warning': return '#ffeedb';
+      case 'danger': return '#fac8df';
+      default: return '#FFF';
+    }
+  } else {
+    switch (type) {
+      case 'success': return '#3c763d';
+      case 'info': return '#31708f';
+      case 'warning': return '#8a6d3b';
+      case 'danger': return '#c20101';
+      default: return '#111';
+    }
+  }
+};
+
+const Alert = ({
+  type = 'default', variant = 'filled', id, className = '', title, description,  allowDismissal = false,
+  onDismiss = () => {}, primaryButton, secondaryButton, titleIcon,
+}: Alert) => {
+  const [hide, setHide] = useState(false);
+  const { isDarkTheme } = useContext(DSRContext);
+  const computedClassName = clsx([
+    description ? 'dsr-py-4' : 'dsr-py-3',
+    'dsr-relative dsr-rounded-lg dsr-px-3 dsr-flex dsr-flex-col dsr-gap-2 ',
+    className,
+  ]);
+
+  return !hide ? (
+      <div
+          id={id}
+          className={computedClassName}
+          style={{
+            color: getColorByType(type, isDarkTheme),
+            borderColor: variant === 'outline' ? getColorByType(type, isDarkTheme ) : undefined,
+            borderStyle: variant === 'outline' ? 'solid' : undefined,
+            borderWidth: variant === 'outline' ? '1px' : undefined,
+            backgroundColor: variant === 'filled' ? getBgByType(type, isDarkTheme) : undefined,
+          }}
+      >
+          {allowDismissal && (
+              <div className="dsr-absolute dsr-top-0 dsr-right-0 dsr-pr-3 dsr-pt-2">
+                  <button
+                      title="dismiss"
+                      className="dsr-font-mono dsr-outline-none dsr-font-bold dsr-text-2xl"
+                      onClick={() => { setHide(true); onDismiss(); }}
+                  >
+                      <Icon icon="times" size={16} />
+                  </button>
+              </div>
+          )}
+          {title && (
+              <h4 className={description ? 'dsr-text-xl dsr-font-semibold' : 'dsr-text-lg'}>
+                  {titleIcon ? <Icon icon={titleIcon} /> : null}
+                  {title}
+              </h4>
+          )}
+          {description && <p className="dsr-text-lg">{description}</p>}
+          {(primaryButton || secondaryButton) && (
+              <div className="dsr-flex dsr-items-center">
+                  {primaryButton &&
+                      <div className="dsr-mr-2">
+                          <Button
+                              {...primaryButton}
+                              color="primary"
+                              className="dsr-text-base"
+                          />
+                      </div>
+                  }
+                  {secondaryButton &&
+                      <div>
+                          <Button
+                              {...secondaryButton}
```

Comment:
I am gonna merge this now, but usually when you do conditions use parenthesis while printing out the div. eg: it should be 

```jsx
{secondaryButton && (
    <div>...</div>
)}
```